weaviate/backup/backup.py

```diff
@@ -21,6 +23,283 @@
+    """Which backend should be used to write the backup to."""
+
+    FILESYSTEM = "filesystem"
+    S3 = "s3"
+    GCS = "gcs"
+    AZURE = "azure"
+
+
+class _BackupStatus(str, Enum):
+    STARTED = "STARTED"
+    TRANSFERRING = "TRANSFERRING"
+    TRANSFERRED = "TRANSFERRED"
+    SUCCESS = "SUCCESS"
+    FAILED = "FAILED"
+
+
+class _BackupStatusReturn(BaseModel):
+    status: _BackupStatus
+    path: str
+
+
+class _BackupReturn(_BackupStatusReturn):
+    collections: List[str] = Field(default_factory=list, alias="classes")
+
+
+class _Backup:
+    """Backup class used to schedule and/or check the status of a backup process of Weaviate objects."""
+
+    def __init__(self, connection: Connection):
+        self._connection = connection
+
+    def create(
+        self,
+        backup_id: str,
+        backend: BackupStorage,
+        include_collections: Optional[Union[List[str], str]] = None,
+        exclude_collections: Optional[Union[List[str], str]] = None,
+        wait_for_completion: bool = False,
+    ) -> _BackupReturn:
+        """Create a backup of all/per collection Weaviate objects.
+
+        Parameters
+        ----------
+        backup_id : str
+            The identifier name of the backup.
+            NOTE: Case insensitive.
+        backend : BackupStorage
+            The backend storage where to create the backup.
+        include_collections : Union[List[str], str], optional
+            The collection/list of collections to be included in the backup. If not specified all
+            collections will be included. Either `include_collections` or `exclude_collections` can be set. By default None.
+        exclude_collections : Union[List[str], str], optional
+            The collection/list of collections to be excluded in the backup.
+            Either `include_collections` or `exclude_collections` can be set. By default None.
+        wait_for_completion : bool, optional
+            Whether to wait until the backup is done. By default False.
+
+        Returns
+        -------
+         A `_BackupReturn` object that contains the backup creation response.
+
+        Raises
+        ------
+        requests.ConnectionError
+            If the network connection to weaviate fails.
+        weaviate.UnexpectedStatusCodeException
+            If weaviate reports a none OK status.
+        TypeError
+            One of the arguments have a wrong type.
+        """
+        (
+            backup_id,
+            backend,
+            include_classes,
+            exclude_classes,
+        ) = _get_and_validate_create_restore_arguments(
+            backup_id=backup_id,
+            backend=backend,  # can be removed when we remove the old backup class
+            include_classes=include_collections,
+            exclude_classes=exclude_collections,
+            wait_for_completion=wait_for_completion,
+        )
+
+        payload = {
+            "id": backup_id,
+            "config": {},
```

Comment:
What is the this `config` option? I don't see it in the docs

Reply:
I actually don't know 🤷  - I copied the existing implementation and changed everything that is necessary (so we can just delete the old stuff after we are done)